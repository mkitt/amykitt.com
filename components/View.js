// @flow
import React from 'react'

export default (props: any) => (
  <div {...props} />
)
