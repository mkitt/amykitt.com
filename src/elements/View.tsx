import styled from '@emotion/styled/macro'
import { memo } from 'react'
import useCss, { useCssType } from '../hooks/useCss'

type PropsType = {
  readonly as?: 'div' | 'main'
} & useCssType

const View = styled.div<PropsType>({}, useCss)

export default memo(View)
