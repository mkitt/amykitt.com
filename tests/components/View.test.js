import React from 'react'
import renderer from 'react-test-renderer'
import View from '../../components/View'

describe('View', () => {
  let component

  beforeEach(() => {
    component = (
      <View data-stuff="view" />
    )
  })

  it('shows the correct snapshot for the View component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
