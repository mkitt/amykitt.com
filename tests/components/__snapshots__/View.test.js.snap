// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View shows the correct snapshot for the View component tree 1`] = `
<div
  data-stuff="view"
/>
`;
